registry/api/v2/urls_test.go

```diff
@@ -174,6 +174,24 @@ func TestBuilderFromRequest(t *testing.T) {
 	forwardedHostHeader2 := make(http.Header, 1)
 	forwardedHostHeader2.Set("X-Forwarded-Host", "first.example.com, proxy1.example.com")
 
+	forwardedHostHeader3 := make(http.Header, 1)
+	forwardedHostHeader3.Set("X-Forwarded-Host", "first.example.com:443")
+
+	forwardedPortHeader := make(http.Header, 1)
+	forwardedPortHeader.Set("X-Forwarded-Port", "5000")
+
+	forwardedPort2Header := make(http.Header, 1)
+	forwardedPort2Header.Set("X-Forwarded-Port", "443 , 5001")
+
+	forwardedComboHeader := make(http.Header, 3)
+	forwardedComboHeader.Set("X-Forwarded-Proto", "https")
+	forwardedComboHeader.Set("X-Forwarded-Host", " first.example.com ")
+	forwardedComboHeader.Set("X-Forwarded-Port", " 12345 \t")
+
+	forwardedMixedHeader := make(http.Header, 2)
+	forwardedMixedHeader.Set("X-Forwarded-Host", "first.example.com:5000")
```

Comment:
This example concerns me. We clearly have a declared port in the host header but we now trust an unbound port? How do we know the same proxy added both headers?